### ISSUE TITLE
Allow TemplateDefinitionManager to redefine templates on the fly

### ISSUE THREAD

**Issue by mtozlu** [abpframework/abp#9590]
In our system, we are using Templating engine of abp v4.4
Our templates are stored in database instead of Virtual File System. So we have the ability to update/add templates on the fly.

However after adding/updating templates, we need to call `Define` method of `TemplateDefinitionProvider` to add new templates. This is done in `TemplateDefinitionManager`. https://github.com/abpframework/abp/blob/dd02ffe0f62807ae543e7bd9e18559d319afa0f8/framework/src/Volo.Abp.TextTemplating.Core/Volo/Abp/TextTemplating/TemplateDefinitionManager.cs#L54. 

So we need to have a way to refresh the definitions. Unfortunately the accessor of this method is `protected`. Is there any reason behind this? If there's no problem making it `public`, I can create a pull-request for that.

That way we will safely call `TemplateDefinitionManager.CreateTextTemplateDefinitions` to refresh definitions after templates are updated/added in database.

**Comment by NecatiMeral** [comment]
You could create a own implementation of `TemplateDefinitionManager` or inherit from `TemplateDefinitionManager`.

**Comment by mtozlu** [comment]
I tried to inherit from `TemplateDefinitionManager` and create a new method for refreshing Templates, however Templates property has no setter, so I had to hide inherited property and use `Templates` property with a private setter. Secondly there is no `RefreshTextTemplateDefinitions` method so I had to create that method. Bu registering this new Manager into DI was a bit problematic because this new refresh method has to be defined in an interface. And I had errors when calling refresh method.

I will try working on this, however I think it would be best if base `TemplateDefinitionManager` had a `Refresh` method ready. If you say it can be, I can create a PR asap.
